### PR TITLE
[Merged by Bors] - chore: unused arguments

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: Mathlib/Algebra/GeomSum.lean [hunk 1]
```diff
@@ -132,7 +132,7 @@ theorem geom_sum₂_self {α : Type*} [CommRing α] (x : α) (n : ℕ) :
         ∑ i ∈ Finset.range n, x ^ (i + (n - 1 - i)) := by
       simp_rw [← pow_add]
     _ = ∑ _i ∈ Finset.range n, x ^ (n - 1) :=
-      Finset.sum_congr rfl fun i hi =>
+      Finset.sum_congr rfl fun _ hi =>
         congr_arg _ <| add_tsub_cancel_of_le <| Nat.le_sub_one_of_lt <| Finset.mem_range.1 hi
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/GeomSum.lean:136]: There's a number of changes like this where you keep a `hi` that's now about an inaccessible variable, which I think people previously weren't excited about, but if core is going to warn on them, I guess we need to.

**Review by Ruben-VandeVelde — APPROVED** [review]
Seems to do what it says on the tin

File: Mathlib/Algebra/GeomSum.lean (hunk 1 above)
Comment (jcommelin) [reply Mathlib/Algebra/GeomSum.lean:136]: I think this is a valid point. Would it be possible to replace `i` with `_i` in cases like this? Might be hard to automate...
I guess `hi` could also become `h`. But that seems even harder to automate.

The rule might be: if `i` is unused, replace it with `_`, unless there is a named-and-used variable that depends on it, in which case, replace with `_i`.

**Comment by fpvandoorn** [comment]
Spot checking looks good.

bors merge
bors d+

**Pull request closed without merging**